test/bdd/pkg/rp/rp_steps.go

```diff
@@ -10,56 +10,80 @@ import (
 	"github.com/cucumber/godog"
+	"github.com/google/uuid"
 	"github.com/ory/hydra-client-go/client"
 	"github.com/ory/hydra-client-go/client/admin"
 	"github.com/ory/hydra-client-go/models"
+	"github.com/trustbloc/edge-core/pkg/log"
 	trustblocvdri "github.com/trustbloc/trustbloc-did-method/pkg/vdri/trustbloc"
+	"golang.org/x/oauth2"
 
 	"github.com/trustbloc/edge-adapter/pkg/restapi/rp/operation"
+	"github.com/trustbloc/edge-adapter/test/bdd/pkg/agent"
 	"github.com/trustbloc/edge-adapter/test/bdd/pkg/bddutil"
 	bddctx "github.com/trustbloc/edge-adapter/test/bdd/pkg/context"
 )
 
 const (
-	rpAdapterURL = "http://localhost:8070"
-	resolverURL  = "http://localhost:8072/1.0/identifiers"
-	hydraURL     = "https://localhost:4445"
+	rpAdapterURL   = "http://localhost:8070"
+	resolverURL    = "http://localhost:8072/1.0/identifiers"
+	hydraAdminURL  = "https://localhost:4445/"
+	hydraPublicURL = "https://localhost:4444/"
 )
 
+var logger = log.New("edge-adapter/bddtests/rp")
+
 type tenantContext struct {
 	*operation.CreateRPTenantResponse
-	callback string
+	callback     string
+	pdHandle     string
+	browser      *http.Client
+	walletConnID string
 }
 
 // Steps is the BDD steps for the RP Adapter BDD tests.
 type Steps struct {
-	context   *bddctx.BDDContext
-	tenantCtx map[string]*tenantContext
+	context    *bddctx.BDDContext
+	tenantCtx  map[string]*tenantContext
+	controller *agent.Steps
 }
 
 // NewSteps returns new agent from client SDK.
 func NewSteps(ctx *bddctx.BDDContext) *Steps {
 	return &Steps{
-		context:   ctx,
-		tenantCtx: make(map[string]*tenantContext),
+		context:    ctx,
+		tenantCtx:  make(map[string]*tenantContext),
+		controller: agent.NewSteps(ctx),
 	}
 }
 
 // RegisterSteps registers agent steps.
 func (s *Steps) RegisterSteps(g *godog.Suite) {
+	g.Step(`^the "([^"]*)" is running on "([^"]*)" port "([^"]*)" with controller "([^"]*)"$`, s.registerAgentController)
 	g.Step(`^a request is sent to create an RP tenant with label "([^"]*)" and callback "([^"]*)"$`, s.createTenant)
 	g.Step(`^the trustbloc DID of the tenant with label "([^"]*)" is resolvable$`, s.resolveDID)
 	g.Step(`^the client ID of the tenant with label "([^"]*)" is registered at hydra$`, s.lookupClientID)
+	g.Step(`^a registered rp tenant with label "([^"]*)" and callback "([^"]*)"$`, s.registerTenantFlow)
+	g.Step(`^the rp tenant "([^"]*)" redirects the user to the rp adapter$`, s.redirectUserToAdapter)
+	g.Step(`the rp adapter "([^"]*)" submits a CHAPI request to "([^"]*)" with presentation-definitions and a didcomm invitation to connect`, s.sendCHAPIRequestToWallet) //nolint:lll
+	g.Step(`^"([^"]*)" accepts the didcomm invitation$`, s.walletAcceptsDIDCommInvitation)
+	g.Step(`^"([^"]*)" connects with the RP adapter "([^"]*)"$`, s.validateConnection)
+}
+
+func (s *Steps) registerAgentController(agentID, inboundHost, inboundPort, controllerURL string) error {
+	return s.controller.ValidateAgentConnection(agentID, inboundHost, inboundPort, controllerURL)
```

Comment:
Can't we use the step definition from agent_controller? This just looks to be passthrough with no extra logic.

Reply:
It's a passthrough unto a different, internal instance. These tests are not interested in registering this wallet agent in the other, "global" instance of the agent controller steps.